.github/workflows/actions.yml

```diff
@@ -23,11 +23,47 @@ jobs:
         with:
           profile: minimal
       - name: Build
-        run: cargo build --verbose --all-features
+        run: cargo build --release --verbose --all-features
         env:
           RUSTFLAGS: -D warnings
       - name: Run tests
         run: cargo test --all --verbose --all-features
+      - name: Create artifacts directory
+        run: mkdir artifacts
+      - name: Copy binaries to a separate directory
+        run: find target/release -maxdepth 1 -type f -executable | xargs --replace cp '{}' artifacts
+      - name: Strip binaries of debugging symbols
+        run: strip artifacts/*
+      - name: Upload binaries for use in a dependent job
+        uses: actions/upload-artifact@v2
+        with:
+          name: client-services
+          path: artifacts/*
+
+
+  upload-binaries-to-gcs:
+    needs: build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: client-services
+          path: artifacts
+      - id: 'auth'
+        uses: 'google-github-actions/auth@v0'
+        with:
+          workload_identity_provider: 'projects/281042598092/locations/global/workloadIdentityPools/github-actions/providers/google-cloud'
+          service_account: 'github-actions@radicle-services.iam.gserviceaccount.com'
+      - name: Upload binaries to Google Cloud Storage
+        if: ${{ github.ref == 'refs/heads/master' }}
+        uses: 'google-github-actions/upload-cloud-storage@v0'
+        with:
+          path: artifacts
+          destination: radicle-client-services/
```

Comment:
Could we maybe store them under `radicle-client-services/main/x86_64-linux/` so it is more clear that they're coming from the `main` branch and what arch they're for?

Reply:
It could be also useful if we could store a copy the same artefacts by commit sha, so that we can try out a specific version. Something like `radicle-client-services/<COMMIT_SHA>/x86_64-linux/`. And limit the gcp retention to something like a couple of GB.

Reply:
Can we first merge this PR?  It's been tested already and proven to work.  I'd rather extend the scope of a task in a separate PR.